app/knight.py

```diff
@@ -0,0 +1,81 @@
+from __future__ import annotations
+from typing import Union
+
+from app.equipment.armour import Armour
+from app.equipment.weapon import Weapon
+from app.equipment.potion import Potion
+
+
+class Knight:
+    def __init__(self, name: str, power: int, hp: int) -> None:
+        self.name = name
+        self.power = power
+        self.hp = hp
+
+        self.armour = []
+        self.weapon = None
+        self.potion = None
+
+        self.attack = self.power
+        self.defence = 0
+
+    def change_weapon(self, weapon_stat: dict) -> None:
+        weapon = Weapon(
+            weapon_stat.get("name"),
+            weapon_stat.get("power"),
+        )
+        self.weapon = weapon
+        self.power += weapon.power
+
+    def add_armour(self, armour: Union[list[Armour], Armour]) -> None:
+        for part in armour:
+            self.armour.append(part.get("part"))
+            self.defence += part.get("protection")
+
+    def add_potion(self, potion: dict) -> None:
+        if potion:
+            name, effect = potion.values()
+            self.potion = Potion(name, effect)
+
+    def drink_potion(self) -> None:
+        if self.potion:
+            for effect in self.potion.effects:
+                match effect.attr:
```

Comment:
match case is good for dynamic unpacking compare operations, there you're trying to apply it like a c-style switch case. Use if-else blocks here, you will remove redundant 1 indent and increase readability